sources/packages/backend/apps/api/src/services/application-change-request/application-change-request.service.ts

```diff
@@ -0,0 +1,183 @@
+      applicationId,
+      { allowEdited: true },
+    );
+    if (!application) {
+      throw new CustomNamedError(
+        `Application ${applicationId} to assess change not found.`,
+        APPLICATION_NOT_FOUND,
+      );
+    }
+    if (
+      application.applicationEditStatus !==
+      ApplicationEditStatus.ChangePendingApproval
+    ) {
+      throw new CustomNamedError(
+        `Application ${applicationId} to assess change not in valid status to be updated.`,
+        INVALID_APPLICATION_EDIT_STATUS,
+      );
+    }
+    const auditUser = { id: auditUserId } as User;
+    const currentDate = new Date();
+    await this.dataSource.transaction(async (transactionalEntityManager) => {
+      // Save the note.
+      await this.noteSharedService.createStudentNote(
+        application.student.id,
+        NoteType.Application,
+        note,
+        auditUserId,
+        transactionalEntityManager,
+      );
+      const applicationRepo =
+        transactionalEntityManager.getRepository(Application);
+      if (applicationEditStatus === ApplicationEditStatus.ChangeDeclined) {
+        // Update the application edit status.
+        const updateResult = await applicationRepo.update(
+          {
+            id: applicationId,
+            applicationEditStatus: ApplicationEditStatus.ChangePendingApproval,
+          },
+          {
+            applicationEditStatus: ApplicationEditStatus.ChangeDeclined,
+            modifier: auditUser,
+            updatedAt: currentDate,
+          },
+        );
+        if (!updateResult.affected) {
+          throw new Error(
+            `Application ${applicationId} to assess change not found or not in valid status to be updated.`,
+          );
+        }
+        // End the workflow.
+        await this.workflowClientService.sendApplicationChangeRequestStatusMessage(
+          applicationId,
+          applicationEditStatus,
+        );
+        return;
+      }
+      const changeRequestApplication = await this.applicationRepo.findOne({
+        select: {
+          id: true,
+          applicationEditStatus: true,
+          precedingApplication: {
+            id: true,
+            currentAssessment: { id: true },
+          },
+          currentAssessment: {
+            id: true,
+          },
+        },
+        relations: {
+          currentAssessment: true,
+          precedingApplication: {
+            currentAssessment: true,
+          },
+        },
+        where: {
+          id: applicationId,
+        },
+      });
+      const previousCompletedApplication =
+        changeRequestApplication.precedingApplication;
+      const previousCompletedApplicationId = previousCompletedApplication.id;
+      const previousCompletedApplicationCurrentAssessment =
+        previousCompletedApplication.currentAssessment;
+      const newApplicationCurrentAssessment =
+        changeRequestApplication.currentAssessment;
```

Comment:
The idea of these "shortcuts" is to improve readability but when we are only concatenation two names and removing the ".", it would not reach the goal.
For instance, the difference between the two below is just a ".": 
- `previousCompletedApplication.currentAssessment`
- `previousCompletedApplicationCurrentAssessment`